### PR TITLE
Added Eclipse IDE paths to .gitignore. Improved CONTRIBUTING doco.

### PR DESCRIPTION
**Pull request by johanssj**
Pretty much as per subject line + organised .gitignore in sections using project lerna as a guide.

The information in CONTRIBUTING.md re where to find build files was out of date and misleading for newbees to the project. Corrected & expanded upon this paragraph after spending too long trying to follow the legacy instructions.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,28 @@
...
+nbproject
+.~lock.*
+.buildpath
+.idea
+composer.lock
+*.sublime-workspace
+*.swp
+*.swo
+/.settings/
+/.project
```
Comment (danez) [inline .gitignore]: I'm personally not a fan of adding environment specific ignores to projects. Imho this should be done in the environment itself with a global gitignore. 
Also compose.lock will never appear, as we are not a PHP project ;)

File: packages/babel-types/README.md [hunk 2]
```diff
@@ -292,6 +292,14 @@ Aliases: `Flow`, `FlowDeclaration`, `Statement`, `Declaration`
...
  - `body` (required)
 
+### t.declareModuleExports(typeAnnotation)
+
+See also `t.isDeclareModuleExports(node, opts)` and `t.assertDeclareModuleExports(node, opts)`.
+
+Aliases: `Flow`, `FlowDeclaration`, `Statement`, `Declaration`
+
+ - `typeAnnotation` (required)
+
```
Comment (danez) [inline packages/babel-types/README.md]: Thanks, I completely forgot to add that in my PR 👍

File: .gitignore (hunk 1 above)
Comment (johanssj) [inline .gitignore]: Thanks for feedback. Actually I agree with you but was unaware of global gitignore. Also I thought that lerna people are gurus compared to me so just followed what they did. Feel free to remove compose.lock as this was copy & paste from lerna's .gitignore or discard my .gitignore changes and just leave .gitignore in some "canonical state".

Comment (johanssj) [inline .gitignore]: If we go the route of not adding environment specific ignores to the project, can we please add something in CONTRIBUTING.md to reflect project policy & modus operandi with global gitignore.

File: packages/babel-types/README.md (hunk 2 above)
Comment (johanssj) [inline packages/babel-types/README.md]: No worries; noticed that, while the file contains a generated section, it contains a human-authored header so thought that the appropriate course of action was to push it upstream.
btw. Just familiarising myself with the Babel neighbourhood so just followed you here on GitHub and on Twitter.

**Comment by hzoo** [comment]
I can merge the babel-types/README.md change. I don't think we need to update .gitignore like danez mentioned, and contributing info can go in the babel-cli readme or the readme itself if you want to submit that.

If you want to have multiple PRs, I would recommend making a branch off of master instead of using your master branch

**Comment by johanssj** [comment]
Agree .gitignore stuff better done like danez suggested. Note that information in contributing.md is just plain wrong though so someone needs to fix that regardless. I don't understand why you can't just cherry pick from my submission whatever you think is relevant when you merge.

**Comment by hzoo** [comment]
Yeah for sure, we can do that if we merge locally rather than the web ui

**Comment by johanssj** [comment]
Sorry I'm still getting my head around this git stuff. Unfortunately the last crowd I worked for was still shoving Clearcase and SVN down our throats. Plus add to that getting acquainted with this project and the way you guys like to do things etc etc etc.
I wouldn't recommend burying stuff one needs to get going with if they want to start contributing to the project too deep. I already mentioned burning time getting started so I've given some suggestions but people who know the project better than me are best off updating documentation and/or stealing from my tiny contribution.

**Comment by hzoo** [comment]
What do you mean by

> I wouldn't recommend burying stuff one needs to get going with if they want to start contributing to the project too deep

The suggestion about using a different branch other than `master` is a general tip for any of the open source projects you might contribute to on github.

I can also edit your branch and merge

**Comment by hzoo** [comment]
Merged manually in https://github.com/babel/babel/commit/6d7417ba1b257e7d921d8f643e1a3d364161d36d

**Comment by johanssj** [comment]
What I mean is for anyone new coming to the project and wanting to contribute, everything that they need to get going with ought be in CONTRIBUTING.md which is in the project root directory, even if it just a link(s) to some other document(s) that reside below the project root, such as in this case, say, a link to the README in babel-cli. Just anything that reduces the barrier to entry is the way to go imho.
Thanks for the branching suggestion (as a general tip) and also your tip to hook up on slack; I'll get the hang of things eventually.
Please let me know when I can close the pull request on my side. If you can copy & paste any words that you think are useful from my pull request in this instance that would be great .. that way I can start afresh on any contributions in the future.

**Comment by johanssj** [comment]
Okay you closed the pull request while I was typing that last para. Sweet.

**Comment by hzoo** [comment]
I merged it but it says closed. If I don't merge on the website and I do it manually it "looks" like it's closed.

**Comment by johanssj** [comment]
This is all really good thanks -- saw the updated readme #packages links and followed thru to babel-cli readme where you borrowed some of my words -- excellent.
Getting a better hang of the routine now so starting over with a fresh fork and fresh branch(s) so I don't go dizzy.
Cheers, Justin.

**Pull request closed without merging**